### DISCUSSION TITLE
Is there a way to add Data to <Handle /> so that I can use that data object in creating an Edge?

### DISCUSSION THREAD

**Discussion by kvedula18** [bcakmakoglu/vue-flow#1058] [Q&A]
I've created a custom node, that presents multiple Handles based on the possible options for the node Yes, No, May be.
I'm creating an edge and adding a label based on the source Handle now, but would like to use the data to determine what label to put, based on the data of the handle. I see data can be added nodes and edges, but not handles.

Am I missing something? Is there another way to achieve the same.
```

export default class SpecialNode extends Vue {
    edges:[] = [{
        id: "a", style: "top:10%", label:"Yes",
    },{
        id: "b", style: "top:25%", label:"No",
    },{
        id: "c", style: "top:50%", label:"May be",
    },{
        id: "d", style: "top:75%", label:"NOTA",
    },
    ]
}
</script>

<template>
    <div >
        Special Node
    </div>
    <Handle id="t" type="target" position="left"/>
    <div v-for="(item, index) in edges" v-bind:key="index">
        <Handle  :id="item.id" 
        type="source" position="right"
        :data="item"
        connectable="single"
        :style="item.style" />
    </div>
</template>
 <style scoped>
 </style>
```
 
 
 and the onConnect is 
 ```

 onConnect((params) => {
  console.log(params);
  addEdges([
    {
      ...params,
      type: "smoothstep",
      markerEnd: MarkerType.ArrowClosed,
      updatable: true,
      label: params.sourceHandle ? params.sourceHandle : "",
    },
  ]);
});
```

Also, is there a way to get details of "source" and "target" nodes from the params? I only get the ids

**Comment by bcakmakoglu** [comment]
Handles have no state object that you could pass data to, hence you can't store any data on them.
You could implement your own custom handle component wrapper and implement some state in there but the library can't help you out here
